soc/arm/atmel_sam/same70/soc.c

```diff
@@ -71,7 +71,7 @@ static ALWAYS_INLINE void clock_init(void)
 	 * by a previous program i.e. bootloader
 	 */
 
-	if (PMC->PMC_SR & PMC_SR_MOSCRCS) {
+	if (!(PMC->CKGR_MOR & CKGR_MOR_MOSCSEL_Msk)) {
```

Comment:
As it is a single bit, I think you should use `CKGR_MOR_MOSCSEL` instead of `CKGR_MOR_MOSCSEL_Msk`.

Reply:
I had noticed that CLOCK_MOR_MOSCSEL is old style mask definition. So I prefer _Msk.
```
#define CKGR_MOR_MOSCSEL                    CKGR_MOR_MOSCSEL_Msk                           /**< \deprecated Old style mask definition for 1 bit bitfield. Use CKGR_MOR_MOSCSEL_Msk instead */
```